internal/impl/confluent/sr/client.go

```diff
@@ -15,33 +15,45 @@
 package sr
 
 import (
-	"bytes"
 	"context"
 	"crypto/tls"
-	"encoding/json"
-	"errors"
 	"fmt"
-	"io"
 	"io/fs"
+	"net"
 	"net/http"
 	"net/url"
-	"regexp"
+	"slices"
+	"time"
 
-	"github.com/redpanda-data/benthos/v4/public/service"
-)
+	"github.com/twmb/franz-go/pkg/sr"
 
-var (
-	escapedSepRegexp = regexp.MustCompile("(?i)%2F")
+	"github.com/redpanda-data/benthos/v4/public/service"
 )
 
 // Client is used to make requests to a schema registry.
 type Client struct {
 	SchemaRegistryBaseURL *url.URL
-	client                *http.Client
+	clientSR              *sr.Client
 	requestSigner         func(f fs.FS, req *http.Request) error
 	mgr                   *service.Resources
 }
 
+type roundTripper struct {
+	reqSigner func(req *http.Request) error
+	*http.Transport
+}
+
+func (rt *roundTripper) RoundTrip(req *http.Request) (*http.Response, error) {
+	// This is naughty, but it's probably fine...
```

Comment:
Is Travis up for allowing some middleware option that intercepts the http request properly?

Reply:
I did ask, but he hasn't gotten around to it yet... Not sure if we should block this PR until that happens, but yeah, it would be great to avoid doing this.